src/Kernels/TimeCommon.cpp

```diff
@@ -39,121 +41,142 @@
  **/
 
 #include "TimeCommon.h"
+#include <DataTypes/ConditionalTable.hpp>
+#include <Initializer/BasicTypedefs.hpp>
+#include <Kernels/Time.h>
+#include <Kernels/precision.hpp>
+#include <Parallel/Runtime/Stream.hpp>
+#include <cassert>
 #include <stdint.h>
+#include <tensor.h>
 
-void seissol::kernels::TimeCommon::computeIntegrals(Time& i_time,
-                                                    unsigned short i_ltsSetup,
-                                                    const FaceType i_faceTypes[4],
-                                                    const double i_currentTime[5],
-                                                    double i_timeStepWidth,
-                                                    real * const i_timeDofs[4],
-                                                    real o_integrationBuffer[4][tensor::I::size()],
-                                                    real * o_timeIntegrated[4] )
-{
+#ifdef ACL_DEVICE
+#include <DataTypes/ConditionalKey.hpp>
+#include <DataTypes/EncodedConstants.hpp>
+#endif
+
+#ifndef NDEBUG
+#include "Common/constants.hpp"
+#include <cstdint>
+#endif
+
+namespace seissol::kernels {
+
+void TimeCommon::computeIntegrals(Time& time,
+                                  unsigned short ltsSetup,
+                                  const FaceType faceTypes[4],
+                                  const double currentTime[5],
+                                  double timeStepWidth,
+                                  real* const timeDofs[4],
+                                  real integrationBuffer[4][tensor::I::size()],
+                                  real* timeIntegrated[4]) {
   /*
    * assert valid input.
    */
   // only lower 10 bits are used for lts encoding
-  assert (i_ltsSetup < 2048 );
+  assert(ltsSetup < 2048);
 
 #ifndef NDEBUG
   // alignment of the time derivatives/integrated dofs and the buffer
-  for( int l_dofeighbor = 0; l_dofeighbor < 4; l_dofeighbor++ ) {
-    assert( ((uintptr_t)i_timeDofs[l_dofeighbor])          % Alignment == 0 );
-    assert( ((uintptr_t)o_integrationBuffer[l_dofeighbor]) % Alignment == 0 );
+  for (int dofneighbor = 0; dofneighbor < 4; dofneighbor++) {
+    assert(reinterpret_cast<uintptr_t>(timeDofs[dofneighbor]) % Alignment == 0);
+    assert(reinterpret_cast<uintptr_t>(integrationBuffer[dofneighbor]) % Alignment == 0);
   }
 #endif
 
   /*
    * set/compute time integrated DOFs.
    */
-  for( unsigned int l_dofeighbor = 0; l_dofeighbor < 4; l_dofeighbor++ ) {
+  for (int dofneighbor = 0; dofneighbor < 4; ++dofneighbor) {
```

Comment:
Why did we remove the unsigned here?

Reply:
...just a habit. :)

It's more common to stumble over `int` as loop var in such a case—but it won't have a large difference here anyways (it's a more common problem when counting the other way). Especially once optimizations are enabled, probably the loop is just unrolled if it's small enough.